src/axom/primal/operators/evaluate_integral.hpp

```diff
@@ -0,0 +1,176 @@
+ *
+ * \param [in] c the Bezier curve object
+ * \param [in] integrand the lambda function representing the integrand. 
+ * Must accept a 2D point as input, and return a double (scalar field) or 
+ * 2D vector object (vector field).
+ * \param [in] npts the number of quadrature nodes
+ * \return the value of the integral
+ */
+template <class Lambda>
+double evaluate_line_integral(const primal::BezierCurve<double, 2>& c,
+                              Lambda&& integrand,
+                              int npts)
+{
+  // Generate quadrature library, defaulting to GaussLegendre quadrature.
+  //  Use the same one for every curve in the polygon
+  //  Quadrature order is equal to 2*N - 1
+  static mfem::IntegrationRules my_IntRules(mfem::Quadrature1D::GaussLegendre);
+  const mfem::IntegrationRule* quad =
+    &(my_IntRules.Get(mfem::Geometry::SEGMENT, 2 * npts - 1));
+
+  return detail::evaluate_line_integral_component(c, integrand, quad);
+}
+
+/*!
+ * \brief Evaluate an integral across a 2D domain bounded by Bezier curves.
+ *
+ * Assumes that the array of Bezier curves is closed and connected. Will compute
+ * the integral regardless, but the result will be meaningless.
+ * Will eventually incorperate the CurverdPolygon class.
+ * Uses a Spectral Mesh-Free Quadrature derived from Green's theorem, evaluating
+ * the area integral as a line integral of the antiderivative over the curve.
+ *
+ * \param [in] cs the array of Bezier curve objects that bound the region
+ * \param [in] integrand the lambda function representing the integrand. 
+ * Must accept a 2D point as input and return a double
+ * \param [in] npts_Q the number of quadrature points to evaluate the line integral
+ * \param [in] npts_P the number of quadrature points to evaluate the antiderivative
+ * \return the value of the integral
+ */
+template <class Lambda>
+double evaluate_area_integral(const axom::Array<primal::BezierCurve<double, 2>>& cs,
+                              Lambda&& integrand,
+                              int npts_Q,
+                              int npts_P = 0)
+{
+  // Generate quadrature library, defaulting to GaussLegendre quadrature.
+  //  Use the same one for every curve in the polygon
+  static mfem::IntegrationRules my_IntRules(mfem::Quadrature1D::GaussLegendre);
+  const mfem::IntegrationRule *quad_Q, *quad_P;
+
+  // Get the quadrature for the line integral.
+  //  Quadrature order is equal to 2*N - 1
+  quad_Q = &(my_IntRules.Get(mfem::Geometry::SEGMENT, 2 * npts_Q - 1));
+
+  // If the quadrature orders are the same (which they usually are),
+  //  then only create one quadrature within the library and then
+  //  the two pointers can point to the same object. Hopefully improves efficiency.
+  if(npts_P <= 0)
+  {
+    npts_P = npts_Q;
+    quad_P = quad_Q;
+  }
+  else
+  {
+    quad_P = &(my_IntRules.Get(mfem::Geometry::SEGMENT, 2 * npts_P - 1));
+  }
+
+  // Define lower bound as lowest control node for all curves
```

Comment:
Clarity: Please add that we're only looking for the y-coordinate.
E.g.
```suggestion
  // Use minimum y-coordinate of curves as lower bound for integration
```